File: demo/windows/Release/src/LinkDisk/network/AuthManager.java
package LinkDisk.network;

import java.io.*;
import java.util.*;

public class AuthManager {
    private static final String TRUST_FILE = "trusted_devices.dat";
    private Set<String> trustedDevices = new HashSet<>();
    
    public AuthManager() {
        loadTrustedDevices();
    }
    
    public boolean isTrusted(String ip) {
        return trustedDevices.contains(ip);
    }
    
    public void addTrustedDevice(String ip) {
        trustedDevices.add(ip);
        saveTrustedDevices();
        System.out.println("已添加信任设备：" + ip);
    }
    
    public void removeTrustedDevice(String ip) {
        trustedDevices.remove(ip);
        saveTrustedDevices();
        System.out.println("已移除信任设备：" + ip);
    }
    
    public List<String> getAllTrustedDevices() {
        return new ArrayList<>(trustedDevices);
    }
    
    private void loadTrustedDevices() {
        File file = new File(TRUST_FILE);
        if (!file.exists()) {
            return;
        }
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(file))) {
            trustedDevices = (Set<String>) ois.readObject();
            System.out.println("加载信任设备：" + trustedDevices);
        } catch (Exception e) {
            System.out.println("加载信任设备失败，将使用空列表");
            trustedDevices = new HashSet<>();
        }
    }
    
    private void saveTrustedDevices() {
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(TRUST_FILE))) {
            oos.writeObject(trustedDevices);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: linkdisk_flutter/副本lib/network/AuthManager.dart
import 'dart:convert';
import 'dart:io';

class AuthManager {
  // 文件名和 Java 完全一致
  static const String _trustFile = "trusted_devices.dat";
  final Set<String> _trustedDevices = {};

  // 构造函数：加载信任列表
  AuthManager() {
    _loadTrustedDevices();
  }

  // 判断设备是否可信
  bool isTrusted(String ip) {
    return _trustedDevices.contains(ip);
  }

  // 添加信任设备
  void addTrustedDevice(String ip) {
    _trustedDevices.add(ip);
    _saveTrustedDevices();
    print("已添加信任设备：$ip");
  }

  // 移除信任设备
  void removeTrustedDevice(String ip) {
    _trustedDevices.remove(ip);
    _saveTrustedDevices();
    print("已移除信任设备：$ip");
  }

  // 获取所有信任设备（返回 List）
  List<String> getAllTrustedDevices() {
    return List.unmodifiable(_trustedDevices);
  }

  // 加载信任设备（替代 Java 序列化）
  void _loadTrustedDevices() {
    final file = File(_trustFile);
    if (!file.existsSync()) {
      return;
    }

    try {
      final jsonString = file.readAsStringSync();
      final List<dynamic> jsonList = jsonDecode(jsonString);
      _trustedDevices.addAll(List<String>.from(jsonList));
      print("加载信任设备：$_trustedDevices");
    } catch (e) {
      print("加载信任设备失败，将使用空列表");
      _trustedDevices.clear();
    }
  }

  // 保存信任设备（JSON 序列化）
  void _saveTrustedDevices() {
    final file = File(_trustFile);
    try {
      final jsonString = jsonEncode(_trustedDevices.toList());
      file.writeAsStringSync(jsonString);
    } catch (e) {
      print(e);
    }
  }
}